File: 2049-eliminate-maximum-number-of-monsters/eliminate-maximum-number-of-monsters.java
class Solution {
    public int eliminateMaximum(int[] dist, int[] speed) {

        int n = dist.length;
        float [] time = new float[n];
        for(int i=0;i<n;i++){

                time[i] = (float)dist[i]/speed[i];

        }
      
        Arrays.sort(time);
        int cnt = 0;
     
        for(int i=0;i<n;i++){
            if (time[i]>i)
                cnt+=1;
            else
                break;

        }
           
        return cnt;
        
    }
}

File: 2049-eliminate-maximum-number-of-monsters/eliminate-maximum-number-of-monsters.py
class Solution:
    def eliminateMaximum(self, dist: List[int], speed: List[int]) -> int:

        n = len(dist)
        time =[dist[i]/speed[i] for i in range(n)]
        time.sort()
        cnt = 0
        for i in range(n):
            if time[i]>i:
                cnt+=1
            else:
                break
        return cnt
        